### PR TITLE
Updated scheduler applied design changes 

### PR DESCRIPTION
**Pull request by kimeunju108**
**What type of PR is this?**
> /kind feature

**What this PR does / why we need it**: This PR proposes 3 major changes in scheduler as follows:
(1) site resource classification - static & dynamic
(2) cluster informer & event handler
(3) push communication with resource collector

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**: 
This PR includes many parts which are commented out. After testing is completely done, it will be cleaned. 

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

**Comment by jshaofuturewei** [comment]
Please talk to Kevin about the new update of the scheduler crd to start http server.

File: globalscheduler/pkg/scheduler/common/config/config.go [hunk 1]
```diff
@@ -181,3 +185,19 @@ func InitPolicyFromFile(policyFile string, policy *types.Policy) error {
...
+	// HTTP Server
+	HttpAddr   string
+	HttpPort   int
+	APIVersion string
+}
+
+func readConf() *Config {
+	return &Config{
+		HttpAddr:   DefaultString("HttpAddr", "0.0.0.0"),
+		HttpPort:   DefaultInt("HttpPort", 8661),
```
Comment (cathyhongzhang) [inline globalscheduler/pkg/scheduler/common/config/config.go:200]: As we discussed, you will create a new PR to replace this hardcoded part, i.e, get the port from the scheduler controller.

**Review by cathyhongzhang — APPROVED** [review]

**Comment by cathyhongzhang** [comment]
could you fix the conflicts and CI failure?

File: globalscheduler/pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -1140,3 +1154,120 @@ func (sched *Scheduler) UpdateSiteDynamicResource(region string, resource *types
 	err = nil
 	return
 }
+
+//This function updates sites' flavor
+func (sched *Scheduler) UpdateFlavor() (err error) {
```
Comment (jshaofuturewei) [inline globalscheduler/pkg/scheduler/scheduler.go:1159]: Why do we need hard coded flavors here?

File: globalscheduler/pkg/scheduler/router/router.go [hunk 3]
```diff
@@ -87,4 +87,11 @@ var routes = Routes{
 		"/allocations",
 		service.Allocations,
 	},
+	Route{
+		"PushSnapshot",
+		strings.ToUpper("Patch"),
+		//http.MethodPatch,
+		"/globalsheduler/v1/regionresources/regionname",
```
Comment (jshaofuturewei) [inline globalscheduler/pkg/scheduler/router/router.go:94]: According to the implementation in globalscheduler/pkg/scheduler/service/api_push_snapshot.go, the request url is /globalsheduler/v1/regionresources/regionname/[regionName], right?

File: globalscheduler/pkg/scheduler/service/api_push_snapshot.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	}
+	result, err := sched.UpdateSiteDynamicResource(region, &resource)
+	if err != nil {
+		klog.Errorf("Schedule failed!, err: %s", err)
+		utils.WriteFailedJSONResponse(resp, http.StatusInternalServerError, utils.InternalServerWithError(err.Error()))
+		return
+	}
+
+	resourceResp := types.SiteResourceRes{Result: result}
+	resp.WriteHeaderAndEntity(http.StatusCreated, resourceResp)
```
Comment (jshaofuturewei) [inline globalscheduler/pkg/scheduler/service/api_push_snapshot.go:55]: It is an update not creation, using StatusOK in most cases.

**Pull request closed without merging**